shared/packages/worker/src/worker/workers/genericWorker/expectationHandlers/lib/scan.ts

```diff
@@ -654,6 +662,145 @@ export function scanLoudness(
+		progress: number
+	) => void,
+	_logger: LoggerInstance,
+	duration: number
+): CancelablePromise<IframesScanResult> {
+	return new CancelablePromise<IframesScanResult>(async (resolve, reject, onCancel) => {
+		let cancelled = false
+
+		const args = [
+			'-hide_banner',
+			'-select_streams',
+			'v',
+			'-show_frames',
+			'-print_format',
+			'csv',
+			'-show_entries',
+			'frame=pts_time,duration_time,pict_type',
+		]
+
+		args.push(...(await getFFMpegInputArgsFromAccessorHandle(sourceHandle)))
+
+		let ffMpegProcess: ChildProcess | undefined = undefined
+
+		const killFFMpeg = () => {
+			// ensure this function doesn't throw, since it is called from various error event handlers
+			try {
+				ffMpegProcess?.stdin?.write('q') // send "q" to quit, because .kill() doesn't quite do it.
+				ffMpegProcess?.kill()
+			} catch (e) {
+				// This is probably OK, errors likely means that the process is already dead
+			}
+		}
+		onCancel(() => {
+			cancelled = true
+			killFFMpeg()
+			reject('Cancelled')
+		})
+
+		ffMpegProcess = spawn(getFFProbeExecutable(), args, {
+			windowsVerbatimArguments: true, // To fix an issue with ffmpeg.exe on Windows
+		})
+
+		const iframes: number[] = []
+		let distance: number | undefined
+		let prevIframePtsTime: number | undefined
+		let frameDuration: number | undefined
+		let distanceVaries = false
+		let maxDistance = 0
+
+		if (!ffMpegProcess.stdout) {
+			throw new Error('spawned ffmpeg-process stdout is null!')
+		}
+
+		const onError = (err: unknown, context: string | undefined) => {
+			if (ffMpegProcess) {
+				killFFMpeg()
+
+				reject(
+					`Error parsing FFMpeg data. Error: "${err} ${
+						err && typeof err === 'object' ? (err as Error).stack : ''
+					}", context: "${context}" `
+				)
+			}
+		}
+
+		ffMpegProcess.stdout
+			.pipe(csvParser({ headers: false }))
+			.on('data', (data) => {
+				if (cancelled) return
+				const pictType = data[3]
+				if (pictType === 'I') {
+					const ptsTime = parseFloat(data[1])
+					const durationTime = parseFloat(data[2])
+					iframes.push(ptsTime)
+					onProgress(ptsTime / duration)
+					if (prevIframePtsTime !== undefined) {
+						const newDistance = ptsTime - prevIframePtsTime
+						if (distance && Math.abs(newDistance - distance) > EPSILON) {
+							distanceVaries = true
+						}
+						maxDistance = Math.max(maxDistance, newDistance)
+						distance = newDistance
+					}
+					prevIframePtsTime = ptsTime
+					frameDuration = durationTime
+				}
+			})
+			.on('error', onError)
+
+		const onClose = (code: number | null) => {
+			if (cancelled) return
+			if (ffMpegProcess) {
+				ffMpegProcess = undefined
+				if (code === 0) {
+					// success
+					if (frameDuration && Math.abs(maxDistance / frameDuration - 1) < EPSILON) {
+						resolve({
+							type: CompressionType.Intra,
+						})
+					} else if (!distanceVaries && distance) {
+						resolve({
+							type: CompressionType.FixedDistance,
+							distance: distance,
```

Comment:
Perhaps it would be slightly more accurate to not use the last distance here (since we do allow for a small `EPSILON`  difference), but rather `prevIframePtsTime / iframes.length`?